### PR TITLE
Use FInAT elements for node allocation

### LINKED ISSUE #967: Use FInAT elements in function spaces
For the allocation of nodes / degrees of freedom. Requires FInAT/FInAT#20.

### PR DESCRIPTION
**Pull request by miklos1**
Resolves #967. Requires FInAT/FInAT#23. FIAT elements are still used when the dual is required.

### CODE REVIEW

**Comment by miklos1** [comment]
The branch test on Drone fails `tests_firedrake/poisson_bc/test_poisson_bc.py::test_poisson_bc` because the convergence is slightly below the expected. The PR test passed. Is this a known problem?

**Comment by thomasgibson** [comment]
Yes, this has happened to me a few times. It happens a bit sporadically.

File: firedrake/slate/slac/compiler.py [hunk 1]
```diff
@@ -479,7 +479,7 @@ def auxiliary_information(builder):
             # compute its shape to declare as an Eigen::MatrixBase object
             temp = ast.Symbol("C%d" % i)
             V = actee.function_space()
-            node_extent = V.fiat_element.space_dimension()
+            node_extent = V.finat_element.space_dimension()
             dof_extent = np.prod(V.ufl_element().value_shape())
```
Comment (thomasgibson) [inline firedrake/slate/slac/compiler.py]: We noted earlier that is calculation is wrong and is fixed in #1010

**Comment by wence-** [comment]
The failure is because that test is actually random (it makes a random perturbation direction).  We should just relax the tolerance somewhat.

File: firedrake/slate/slac/compiler.py (hunk 1 above)
Comment (miklos1) [inline firedrake/slate/slac/compiler.py]: The point here is to use the FInAT element rather than the FIAT element.

**Comment by miklos1** [comment]
I guess this should be ready to merge then (subject to review).

**Comment by wence-** [comment]
I think this is fine.

**Pull request merged**